### ISSUE TITLE
`filter_genes` fails when `adata.layers['unspliced']` is sparse and `adata.layers['spliced']` dense

### ISSUE THREAD

**Issue by honghh2018** [theislab/scvelo#533]
Thank the tools.
   I preprocess the splice and unsplice in seurat using standard workflow like below:

```R
tm <- as.Seurat(x = ldat)
tm[["RNA"]] <- tm[["spliced"]]
tm <- NormalizeData(tm)
tm <- FindVariableFeatures(tm, selection.method = "vst", nfeatures = 2000)

all.genes <- rownames(tm)
tm <- ScaleData(tm, features = all.genes)

tm <- RunPCA(tm)
ElbowPlot(tm)
tm <- RunUMAP(tm, dims = 1:15)
tm <- FindNeighbors(tm, dims = 1:15)
tm <- FindClusters(tm)
```
and then 
i convert the seurat to h5ad and load it by scv.read("sample.h5ad")
AnnData object with n_obs × n_vars = 42378 × 31053
    obs: 'orig.ident', 'nCount_spliced', 'nFeature_spliced', 'nCount_unspliced', 'nFeature_unspliced', 'nCount_ambiguous', 'nFeature_ambiguous', 'nCount_RNA', 'nFeature_RNA', 'anno', 'initial_size_spliced', 'initial_size_unspliced', 'initial_size'
    var: 'features', 'ambiguous_features', 'unspliced_features'
    obsm: 'X_umap'
    layers: 'ambiguous', 'spliced', 'unspliced'

and the error happen when runing below step:
```python
scv.pp.filter_and_normalize(adata, min_shared_counts=20, n_top_genes=2000)
```

```pytb
ValueError                                Traceback (most recent call last)
<ipython-input-10-5a1623ad56e8> in <module>
----> 1 scv.pp.filter_and_normalize(adata, min_shared_counts=20, n_top_genes=2000)

~/miniconda3/envs/scvelo/lib/python3.8/site-packages/scvelo/preprocessing/utils.py in filter_and_normalize(data, min_counts, min_counts_u, min_cells, min_cells_u, min_shared_counts, min_shared_cells, n_top_genes, retain_genes, subset_highly_variable, flavor, log, layers_normalize, copy, **kwargs)
    799         logg.warn("Could not find spliced / unspliced counts.")
    800 
--> 801     filter_genes(
    802         adata,
    803         min_counts=min_counts,

~/miniconda3/envs/scvelo/lib/python3.8/site-packages/scvelo/preprocessing/utils.py in filter_genes(data, min_counts, min_cells, max_counts, max_cells, min_counts_u, min_cells_u, max_counts_u, max_cells_u, min_shared_counts, min_shared_cells, retain_genes, copy)
    282             Xs, Xu = adata.layers["spliced"], adata.layers["unspliced"]
    283             nonzeros = (
--> 284                 (Xs > 0).multiply(Xu > 0) if issparse(Xs) else (Xs > 0) * (Xu > 0)
    285             )
    286             X = (

~/miniconda3/envs/scvelo/lib/python3.8/site-packages/scipy/sparse/base.py in __rmul__(self, other)
    548             except AttributeError:
    549                 tr = np.asarray(other).transpose()
--> 550             return (self.transpose() * tr).transpose()
    551 
    552     #######################

~/miniconda3/envs/scvelo/lib/python3.8/site-packages/scipy/sparse/base.py in __mul__(self, other)
    514 
    515             if other.shape[0] != self.shape[1]:
--> 516                 raise ValueError('dimension mismatch')
    517 
    518             result = self._mul_multivector(np.asarray(other))

ValueError: dimension mismatch
 #
```

  What happen on this case?
  how to fix it ?

Best,
Hanhuihong

**Comment by WeilerP** [comment]
@honghh2018, which package versions are you using? Please provide the output of `scv.logging.print_versions()`.

*On a general note:* Please do not provide the GitHub issue template if possible and provide all the requested info. This simplifies and speeds up debugging issues immensely.

**Comment by honghh2018** [comment]
Hi @WeilerP , 
      Thank again the reply, 
     The scv.logging.print_version() and scv.logging.print_versions() info lying below:
     Running scvelo 0.2.3 (python 3.8.8) on 2021-07-16 09:16.
     scvelo==0.2.3  scanpy==1.7.1  anndata==0.7.5  loompy==3.0.6  numpy==1.20.1  scipy==1.6.1  matplotlib==3.3.4  sklearn==0.24.1  pandas==1.2.2 
     The adata.layers['unspliced'].shape and adata.layers['unspliced'].shape displaying below:
      (42378, 31053)
      (42378, 31053)
     The type of layers in adata.layers['spliced'] and adata.layers['unspliced'] following:
     numpy.ndarray
     anndata._core.views.SparseCSRView
     Hope the detail can help.
     Let me know if any information not enough to tackle that issue.
     
Best work,
Hanhuihong

**Comment by WeilerP** [comment]
@honghh2018, converting `adata.layer['spliced']` to a sparse matrix in CSR format, i.e.

```
from scipy.sparse import csr_matrix

adata.layers['spliced'] = csr_matrix(adata.layers['spliced'])
```

 should fix the problem.

**Comment by honghh2018** [comment]
Great, it work.

**Comment by WeilerP** [comment]
Can be reproduce with the following dummy example:

```python
import numpy as np
from scipy.sparse import csr_matrix

from anndata import AnnData

import scvelo as scv

adata = AnnData(
    np.eye(2),
    layers={
        'unspliced': csr_matrix(np.eye(2)),
        'spliced': np.eye(2)
    }
)
scv.pp.filter_genes(adata, min_shared_counts=1)
```

**Comment by WeilerP** [comment]
Fixed by #537.
